packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -1250,47 +1245,129 @@ export function makeWallet({
+    },
+    get: petname => installationMapping.petnameToVal.get(petname),
+    getAll: () => installationMapping.petnameToVal.entries(),
+    add: (petname, thing) => addInstallation(petname, thing),
+  });
+
+  const instanceManager = harden({
+    rename: (petname, instance) => {
+      instanceMapping.renamePetname(petname, instance);
+      // We don't wait for the update before returning.
+      updateAllState();
+    },
+    get: petname => instanceMapping.petnameToVal.get(petname),
+    getAll: () => instanceMapping.petnameToVal.entries(),
+    add: (petname, instance) => {
+      // We currently just add the petname mapped to the instance
+      // value, but we could have a list of known instances for
+      // possible display in the wallet.
+      petname = instanceMapping.suggestPetname(petname, instance);
+      // We don't wait for the update before returning.
+      updateAllState();
+    },
+  });
+
+  const issuerManager = harden({
+    rename: (petname, issuer) => {
+      assert(
+        brandTable.hasByIssuer(issuer),
+        `issuer has not been previously added`,
+      );
+      const brandRecord = brandTable.getByIssuer(issuer);
+      brandMapping.renamePetname(petname, brandRecord.brand);
+      updateAllState();
+    },
+    get: petname => {
+      const brand = brandMapping.petnameToVal.get(petname);
+      return brandTable.getByBrand(brand).issuer;
+    },
+    getAll: () => {
+      // TODO: what is the return value?
+      return brandMapping.petnameToVal.entries().map(([petname, brand]) => {
+        const { issuer } = brandTable.getByBrand(brand);
+        return [petname, issuer];
+      });
+    },
+    add: async (petname, issuerP) => {
+      const { brand, issuer } = await brandTable.initIssuer(issuerP);
+      if (!issuerToBoardId.has(issuer)) {
+        const issuerBoardId = await E(board).getId(issuer);
+        issuerToBoardId.init(issuer, issuerBoardId);
+      }
+      brandMapping.suggestPetname(petname, brand);
+      updateAllIssuersState();
+      return issuer;
+    },
+  });
+
+  function getInstallationManager() {
+    return installationManager;
   }
 
-  async function renameInstance(petname, instance) {
-    instanceMapping.renamePetname(petname, instance);
-    await updateAllState();
-    return `instance ${q(petname)} successfully renamed in wallet`;
+  function getInstanceManager() {
+    return instanceManager;
   }
 
-  async function renameInstallation(petname, installation) {
-    installationMapping.renamePetname(petname, installation);
-    await updateAllState();
-    return `installation ${q(petname)} successfully renamed in wallet`;
+  function getIssuerManager() {
+    return issuerManager;
   }
 
-  function getIssuer(petname) {
-    const brand = brandMapping.petnameToVal.get(petname);
-    return brandTable.getByBrand(brand).issuer;
+  async function findAllInvitationAmount(invitationDetailsCriteria) {
+    const zoeInvitationPurse = getPurse(ZOE_INVITE_PURSE_PETNAME);
```

Comment:
Ok, just so I understand, you are thinking that the user, not knowing any better, might manually deposit their Zoe invitations in a second purse and then wonder why the process of making an offer through the wallet isn't finding the invitation?

Hmm, that's a good point, and iterating over the purses of the Zoe invite brand isn't too bad. Let me see what I can do.

Reply:
Hmm, ok, I took another look, and since we are already defining this explicitly in the wallet as a single purse (see `compileOffer` which assumes desirable invitations don't exist in other purses in the same way this method does), I'm thinking of not trying to tackle this issue in my changes today.  I'll make an issue though.